File: TCDNew-master/src/ui.wince/Cen.Wms.Client/Actions/Operation/PurchaseTaskAcceptByTask.cs
using Cen.Wms.Client.Actions.UI.Purchase;
using Cen.Wms.Client.Actions.UI.Utility;
using Cen.Wms.Client.Common;

namespace Cen.Wms.Client.Actions.Operation
{
    class PurchaseTaskAcceptByTask
    {
        public static void Run()
        {
            var purchasesTaskId = PurchaseTaskWait.Run();
            if (purchasesTaskId != null)
            {
                PurchaseTaskContentScan.Run(purchasesTaskId);
            }
            else
                ShowModalMessage.Run(Messages.TitleError, "Задание не выбрано!");
        }
    }
}

File: TCDNew-master/src/ui.android/lib/actions/operation/PurchaseTaskAcceptByTask.dart
import 'package:flutter/material.dart';
import 'package:tiq/actions/ui/purchase/PurchaseTaskContentScan.dart';
import 'package:tiq/actions/ui/purchase/PurchaseTaskWait.dart';
import 'package:tiq/actions/ui/utility/ShowModalMessage.dart';
import 'package:tiq/common/Messages.dart';

class PurchaseTaskAcceptByTask {
  Future<void> run(BuildContext context) async {
    var purchaseTaskId = await PurchaseTaskWait().run(context);
    if (purchaseTaskId != null) {
      await PurchaseTaskContentScan().runWithPurchaseTaskId(context, purchaseTaskId);
    }
    else {
      await ShowModalMessage().run(context, Messages.TitleError, 'Задание не выбрано!');
    }
  }
}
